src/azure-cli/azure/cli/command_modules/storage/_validators.py

```diff
@@ -1146,16 +1146,3 @@ def validator_delete_retention_days(namespace):
     if namespace.enable_delete_retention is False and namespace.delete_retention_days is not None:
         raise ValueError(
             "incorrect usage: '--delete-retention-days' is invalid when '--enable-delete-retention' is set to false")
-
-    if namespace.enable_delete_retention is None and namespace.delete_retention_days is not None:
-        raise ValueError(
-            "incorrect usage: please specify '--enable-delete-retention true' if you want to set the value for "
-            "'--delete-retention-days'")
```

Comment:
Removing the validation of the common call here affects the logic of `update_blob_service_properties()` in this parameter scenario.

Reply:
Yes, I know.  I removed this part because I think we should allow users to update days when delete retention policy is already enabled.
But I think I also need some logic to check inside update_blob_service_properties(). What do you think of such design?

Reply:
I think it's ok, or maybe we can change the update_blob_service_properties logic to be consistent with the file-service-properties.